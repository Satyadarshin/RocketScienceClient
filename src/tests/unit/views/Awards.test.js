/*
 * Authors view unit test with Jest.
 */
import faker from 'faker';
import Vue from 'vue';
import { mount } from '@vue/test-utils'
import AwardsView from '@/views/Awards.vue';

describe('Awards view', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = mount(AwardsView, {
      // Props to pass in for this test
      propsData: { },
      // Stubs allows us to test this component without having to worry about its child components.
      stubs: ['app-award-table', 'md-button'],
    })
  });

  it('Snapshot', ()=> {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should display the main heading', () => {
    // For every unit test there will be an expected result,
    // based on how we expect the code to behave if it is working correctly.
    const expected = 'Science Fiction Literary Awards';
    // An actual result which we get from running the code.
    const actual = wrapper.find('h2').text();
    // We then compare the two to check if the code behaves as expected.
    expect(actual).toEqual(expected);
  });
});