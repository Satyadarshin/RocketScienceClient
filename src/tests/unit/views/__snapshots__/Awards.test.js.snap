// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `undefined`;

exports[`Awards view Snapshot 1`] = `
<div class="md-layout">
  <div class="md-layout-item">
    <h2>Science Fiction Literary Awards</h2>
    <div id="award-category-selector">
      <md-button-stub class="md-raised md-primary selector">Hugo Award for Best Novel</md-button-stub>
      <md-button-stub class="md-raised md-primary selector">Nebula Award for Best Novel</md-button-stub>
    </div>
  </div>
  <div class="md-layout-item">
    <app-award-table-stub who="" what="" selectedaward=""></app-award-table-stub>
  </div>
</div>
`;
